File: src/main/java/scrumweb/dto/projectfield/RadioButtonContainerDto.java
package scrumweb.dto.projectfield;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@NoArgsConstructor
@Getter @Setter
public class RadioButtonContainerDto extends ProjectFieldDto {
    private Set<RadioButtonDto> elements;

    public RadioButtonContainerDto(Long id, String fieldType, String fieldName, Boolean isRequired, Set<RadioButtonDto> elements) {
        super(id, fieldType, fieldName, isRequired);
        this.elements = elements;
    }
}

File: src/frontend/src/app/model/project-fields/RadioButtonContainerDto.ts
import { RadioButtonDto } from './RadioButtonDto';
import { ProjectFieldDto } from './ProjectFieldDto';

export class RadioButtonContainerDto extends ProjectFieldDto {
    elements: Array<RadioButtonDto> = [];

    constructor(id: number, fieldType: string, fieldName: string, isRequired: boolean, elements: Array<RadioButtonDto>) {
        super(id, fieldType, fieldName, isRequired);
        this.elements = elements;
    }
}
